src/modules/hyprland/workspaces.cpp

```diff
@@ -161,14 +202,29 @@ void Workspace::update(const std::string &format, const std::string &icon) {
   Glib::RefPtr<Gtk::StyleContext> style_context = button_.get_style_context();
   add_or_remove_class(style_context, active(), "active");
 
-  label_.set_markup(
-      fmt::format(fmt::runtime(format), fmt::arg("id", id()), fmt::arg("icon", icon)));
+  label_.set_markup(fmt::format(fmt::runtime(format), fmt::arg("id", id()),
+                                fmt::arg("name", name()), fmt::arg("icon", icon)));
 }
 
 void Workspaces::sort_workspaces() {
   std::sort(workspaces_.begin(), workspaces_.end(),
-            [](std::unique_ptr<Workspace> &lhs, std::unique_ptr<Workspace> &rhs) {
-              return lhs->id() < rhs->id();
+            [](std::unique_ptr<Workspace> &a, std::unique_ptr<Workspace> &b) {
+              // normal -> named -> special -> named special
+              if (a->id() > 0 && b->id() > 0) {
+                return a->id() < b->id();
+              }
+              if (a->id() < 0 && b->id() < 0) {
+                if ((a->is_special()) ^ (a->is_special())) {
+                  return a->id() > b->id();
+                } else {
+                  return a->id() < b->id();
+                }
+              }
+              if ((a->id() > 0) ^ (b->id() > 0)) {
+                return a->id() > b->id();
+              }
+              spdlog::error("huh!!?");
```

Comment:
?))

Reply:
should never get there
will probably make it unreachable

Reply:
removed it